var/spack/repos/builtin/packages/py-distributed/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyDistributed(PythonPackage):
+    """Distributed scheduler for Dask"""
+
+    homepage = "https://distributed.dask.org/"
+    url      = "https://pypi.io/packages/source/d/distributed/distributed-2.10.0.tar.gz"
+
+    version('2.10.0', '300e5495d0306f472c1aa34a522273e3066b9f50d3eb25cee17cd3b6177eea54')
+    version('1.28.1', sha256='3bd83f8b7eb5938af5f2be91ccff8984630713f36f8f66097e531a63f141c48a')
+
+    depends_on('python@2.7:2.8,3.5:', when='@:1')
+    depends_on('python@3.5:', when='@2:')
+    depends_on('py-setuptools', type='build')
+
+    depends_on('py-click@6.6:', type=('build', 'run'))
+    depends_on('py-cloudpickle@0.2.2:', type=('build', 'run'))
+    depends_on('py-msgpack', type=('build', 'run'))
+    depends_on('py-psutil@5.0:', type=('build', 'run'))
+    depends_on('py-six', type=('build', 'run'))
```

Comment:
Not needed, at least in the latest version.

Reply:
See https://github.com/dask/distributed/blob/2.10.0/requirements.txt and https://github.com/dask/distributed/blob/1.28.1/requirements.txt . And a simple `pip install distributed==1.28.1 | grep -e 'from distributed'` test confirms that:
```
Collecting pyyaml (from distributed==1.28.1)
Collecting msgpack (from distributed==1.28.1)
Collecting cloudpickle>=0.2.2 (from distributed==1.28.1)
Collecting sortedcontainers!=2.0.0,!=2.0.1 (from distributed==1.28.1)
Collecting singledispatch; python_version < "3.4" (from distributed==1.28.1)
Collecting six (from distributed==1.28.1)
Collecting click>=6.6 (from distributed==1.28.1)
Collecting dask>=0.18.0 (from distributed==1.28.1)
Collecting tornado>=5 (from distributed==1.28.1)
Collecting futures; python_version < "3.0" (from distributed==1.28.1)
Collecting toolz>=0.7.4 (from distributed==1.28.1)
Collecting psutil>=5.0 (from distributed==1.28.1)
Collecting zict>=0.1.3 (from distributed==1.28.1)
Collecting tblib (from distributed==1.28.1)
```

You can get similar results for 2.10.0

Reply:
> See https://github.com/dask/distributed/blob/2.10.0/requirements.txt

six is not listed here. It's needed for 1.28.1 because that version supports both Python 2 and 3, but 2.10.0 only supports Python 3, so six is no longer used. Using your pip method may bring in indirect dependencies.

Reply:
Ah, you only meant `py-six`, not all 4 (`py-cloudpickle`, `py-msgpack`, `py-psutil`).

Reply:
Yep, just commenting on a single line